encoding.bs

```diff
@@ -94,37 +96,177 @@ bitwise AND by "&amp;", and bitwise OR by "|".
 
 <hr>
 
-<p>A <dfn id=concept-token>token</dfn> is a piece of data, such as a <a>byte</a> or
-<a>scalar value</a>.
+<p>An <dfn id=concept-stream export>I/O queue</dfn> is a type of <a>list</a>
+with <a>items</a> of a particular type (i.e. <a>bytes</a> or <a>scalar
+values</a>). <dfn id="end-of-stream" export>End-of-file</dfn> is a special
+<a>item</a> that can be present in <a for=/>I/O queues</a> of any type, and that
+signifies no more <a>items</a> are in the queue.
 
-<p>A <dfn id=concept-stream>stream</dfn> represents an ordered sequence of
-<a>tokens</a>. <dfn>End-of-stream</dfn> is a special
-<a>token</a> that signifies no more
-<a>tokens</a> are in the
-<a for=/>stream</a>.
+<p>To <dfn>create an I/O queue</dfn>, return a new <a for=/>I/O queue</a>
```

Comment:
Shouldn't this also take some data as an argument to put in the queue?

Reply:
Here I was trying to take algorithms defined to take "one or more tokens" and split them into two algorithms that took an item and a sequence. But maybe I should try and merge both overloads into one algorithm.

Reply:
I was thinking of "create an I/O queue" as an operation that would replace the "a new stream" text in the API algorithms: it creates an "empty" immediate queue and leaves it to the calling algorithm to push items into it, sincepush is careful to sidestep the end-of-queue item. But if you think it should take data to initialize the queue, that's fine.

Reply:
It might be simpler to write « end-of-file » if you want that.